### PR TITLE
docs: translate internals/architecture.mdx to pt-br

### PR DESCRIPTION
**Pull request by ImBIOS**
## Summary

Part of:

- #1065

## Test Plan

Please any native speaker, review my translation work.

### CODE REVIEW

**Review by danielsarsi — COMMENTED** [review]
Giving some thoughts. Great work!

**Comment by ImBIOS** [comment]
I revised by making sure the output and some code snippets, using the original (English).

File: website/src/content/docs/pt-br/internals/architecture.mdx [hunk 1]
```diff
@@ -41,21 +41,20 @@ const b = "bar";
   3: EOF@55..55 "" [] []
 ```
 
-The CST is never directly accessible by design, a developer can read its information using the Red tree, using a number of APIs that are autogenerated from the grammar of the language.
+O CST nunca é diretamente acessível por design, um desenvolvedor pode ler suas informações usando a Árvore Vermelha, usando uma série de APIs que são geradas automaticamente a partir da gramática do idioma.
```
Comment (victor-teles) [inline website/src/content/docs/pt-br/internals/architecture.mdx:44]:
```suggestion
O CST nunca é diretamente acessível por design, um desenvolvedor pode ler suas informações usando a Árvore Vermelha, usando uma série de APIs que são geradas automaticamente a partir da gramática da linguagem.
```

File: website/src/content/docs/pt-br/internals/architecture.mdx [hunk 2]
```diff
@@ -41,21 +41,20 @@ const b = "bar";
...
 
 
-#### Resilient and recoverable parser
+#### Parser resiliente e recuperável
 
-In order to construct a CST, a parser needs to be error resilient and recoverable:
-- resilient: a parser that is able to resume parsing after encountering syntax error that belong to the language;
-- recoverable: a parser that is able to **understand** where an error occurred, and being able to resume the parsing by creating **correct** information;
+Para construir um CST, é necessário um parser resiliente e recuperável:
+- resiliente: um parser capaz de retomar a análise após encontrar um erro de sintaxe pertencente ao idioma;
```
Comment (victor-teles) [inline website/src/content/docs/pt-br/internals/architecture.mdx:50]:
```suggestion
- resiliente: um parser capaz de retomar a análise após encontrar um erro de sintaxe pertencente à linguagem;
```

**Review by victor-teles — APPROVED** [review]
Looks good to me!

I left a small suggestion about "language" word translation.

**Review by danielsarsi — APPROVED** [review]
It's great! 👍

**Pull request merged**